rfd/0211-azure-devops-joining.md

```diff
@@ -0,0 +1,521 @@
+---
+authors: Noah Stride (noah@goteleport.com)
+state: draft
+---
+
+# RFD 211 - Azure DevOps Joining
+
+## Required Approvals
+
+- Engineering: TimB && DanU
+- Product: DaveS 
+
+## What
+
+Allow Bots & Agents to secretlessly authenticate to Teleport from Azure DevOps
+pipelines.
+
+## Why
+
+Azure DevOps is a popular CI/CD platform, and today, leveraging Teleport 
+Machine & Workload Identity from Azure DevOps is not possible without
+laborious workarounds.
+
+An Azure Devops join method would allow Bots/Agents to authenticate to Teleport
+without the use of long-lived secrets, and provide richer metadata for audit
+logging & authorization decision purposes.
+
+## Details
+
+Goals:
+
+- Allow authentication to Teleport from Azure DevOps pipelines without the use
+  of long-lived secrets.
+- It should be possible to scope this authentication to a specific Azure DevOps
+  pipeline. 
+- Mitigate common attacks such as token reuse.
+
+This RFD will make reference to and rely on prior art from other OIDC join
+methods e.g `github`, `gitlab`.
+
+### Overview
+
+The Azure DevOps join method will be named `azure_devops` and will be a 
+delegated, non-renewable join method.
+
+The Azure DevOps join method will leverage the OpenID Connect (OIDC) token that
+is available to pipelines via a special internal API. A public OpenID
+configuration document and JWKS are available to support validating these
+tokens.
```

Comment:
Just to check my understanding, am I correct that this is different from the approach suggested in #38852, in that it doesn't use the federation feature and instead validates the using the shared JWKS available at `https://vstoken.dev.azure.com/`?

Reply:
I'd probably argue that it's less a case of "original" token and more a case of a "different" token. https://github.com/gravitational/teleport/issues/38852 leverages the ServiceConnection-specific token, which at that time was the only one that existed. Since then, they've added a token that is specific to the Pipeline rather than a ServiceConnection.